### PR TITLE
Enable counting of articles from FAQ sub-categories for FAQ explorer.

### PR DESCRIPTION
**Pull request by reneeb**
Currently visitors of the FAQ explorer might see the article count 0 if articles only exist
in subcategories. This might people make think that no articles exist at all. And additionally
the ticket overview by queue works differently.

To keep backwards compatibility the new behaviour has to be switched on in the sysconfig.

### CODE REVIEW

**Comment by rkaldung** [comment]
@reneeb Really like this.

**Comment by rkaldung** [comment]
@jepf Do you think we can this push into the upcoming version too? If not let's keep it for the next round.

File: Kernel/Config/Files/XML/FAQ.xml [hunk 1]
```diff
@@ -2781,4 +2781,11 @@ Your OTRS Notification Master
             <Item ValueType="String" ValueRegex="^\d+$">4</Item>
         </Value>
     </Setting>
+    <Setting Name="FAQExplorer::CountSubCategories" Required="0" Valid="1">
+        <Description Translatable="1">Count articles from FAQ subcategories for explorer.</Description>
```
Comment (jepf) [inline Kernel/Config/Files/XML/FAQ.xml]:
```suggestion
        <Description Translatable="1">Enable counting of articles from FAQ sub-categories for FAQ explorer.</Description>
```

Could you please add the German translation for the description to de_FAQ.pm?

File: Kernel/System/FAQ.pm [hunk 2]
```diff
@@ -1202,6 +1202,28 @@ sub FAQCount {
...
+            next if !exists $CategoryList->{$CategoryID};
+
+            push @CategoryIDs, keys %{ $CategoryList->{$CategoryID} };
+        }
+
+        $Param{CategoryIDs} = [
+            keys %AllCategories
+        ];
+    }
+
```
Comment (jepf) [inline Kernel/System/FAQ.pm]:
```suggestion
    my $ConfigObject = $Kernel::OM->Get('Kernel::Config');
    
    my $CountSubCategories = $ConfigObject->Get('FAQExplorer::CountSubCategories');
    if ( $Param{CountSubCategories} && $CountSubCategories ) {
        my %Categories;

        my $CategoryList = $Self->CategoryList(
            UserID => $Param{UserID},
        );

        my @CategoryIDs = @{ $Param{CategoryIDs} || [] };
        CATEGORYID:
        while ( my $CategoryID = shift @CategoryIDs ) {
            $Categories{$CategoryID}++;

            next CATEGORYID if !exists $CategoryList->{$CategoryID};

            push @CategoryIDs, keys %{ $CategoryList->{$CategoryID} };
        }

        $Param{CategoryIDs} = [
            keys %Categories,
        ];
    }
```

Could you please add a small unit test for this?

**Review by jepf — CHANGES_REQUESTED** [review]

**Review by NiklasSchmitt — APPROVED** [review]

**Pull request merged**